### PR TITLE
infra: Add kubevirtci cluster infrastructure

### PR DESCRIPTION
**Pull request by qinqon**
This change integrate a kubevirtci cluster to test
cloud-provider-kubevirt locally.

Extras:
- [x] Move manifest to "config" kustomize directory
- [x] Use kustomize to replace image with kubevirtci registry:5000
- [x] Use kustomize to generate secrets

Job to exercise this PR https://github.com/kubevirt/project-infra/pull/2205

### CODE REVIEW

**Review by nirarg — COMMENTED** [review]
Great job
two small comments, and than I think it good to go

File: Makefile [hunk 1]
```diff
@@ -51,3 +51,19 @@ push:
...
+cluster-up:
+	./kubevirtci up
+
+.PHONY: cluster-sync
+cluster-sync:
+	./kubevirtci sync
+
+.PHONY: cluster-down
+cluster-down:
+	./kubevirtci down
```
[the remark this answers is not in the document]
Comment (qinqon) [reply Makefile:65]: This is common pattern at all kubevirt projects, so people landing here from kubevirt are use to, also is usually good to have make target as the entry point for CI, but is not a must, do I still remove them ?

**Review by rhrazdil — APPROVED** [review]

File: kubevirtci [hunk 2]
```diff
@@ -174,14 +174,7 @@ function kubevirtci::sync() {
...
     REGISTRY="127.0.0.1:$(cluster-up/cluster-up/cli.sh ports registry)" make image push
-
-    cat << EOF > config/secret/cloud-config
-kubeconfig: |
-$($_ssh_infra node01  -- sudo cat  /etc/kubernetes/admin.conf | sed "s/^/  /g")
-loadBalancer:
-  creationPollInterval: 30
-namespace: ${TENANT_CLUSTER_NAMESPACE}
-EOF
+    $_ssh_infra node01  -- sudo cat  /etc/kubernetes/admin.conf | sed "s/^/  /g" > config/secret/infra-kubeconfig
```
Comment (rhrazdil) [inline kubevirtci:177]: is `sed "s/^/ /g"` needed?

**Comment by qinqon** [comment]
/hold
The new e2e prow job is failing.

**Comment by qinqon** [comment]
/hold cancel
e2e job working now, cloud-config file was missing from the PR.

**Comment by qinqon** [comment]
/retest
It's fine locally

**Pull request merged**